chunks/lang-java/Dockerfile

```diff
@@ -8,18 +8,18 @@ USER gitpod
 # Dazzle does not rebuild a layer until one of its lines are changed. Increase this counter to rebuild this layer.
 ENV TRIGGER_REBUILD=1
 
+SHELL [ "/bin/bash", "-c" ]
 RUN curl -fsSL "https://get.sdkman.io" | bash \
- && bash -c ". /home/gitpod/.sdkman/bin/sdkman-init.sh \
-             && sed -i 's/sdkman_selfupdate_enable=true/sdkman_selfupdate_enable=false/g' /home/gitpod/.sdkman/etc/config \
-             && sed -i 's/sdkman_selfupdate_feature=true/sdkman_selfupdate_feature=false/g' /home/gitpod/.sdkman/etc/config \
-             && sdk install java ${JAVA_VERSION} \
-             && sdk install gradle \
-             && sdk install maven \
-             && sdk flush archives \
-             && sdk flush temp \
-             && mkdir /home/gitpod/.m2 \
-             && printf '<settings>\n  <localRepository>/workspace/m2-repository/</localRepository>\n</settings>\n' > /home/gitpod/.m2/settings.xml \
-             && echo 'export SDKMAN_DIR=\"/home/gitpod/.sdkman\"' >> /home/gitpod/.bashrc.d/99-java \
-             && echo '[[ -s \"/home/gitpod/.sdkman/bin/sdkman-init.sh\" ]] && source \"/home/gitpod/.sdkman/bin/sdkman-init.sh\"' >> /home/gitpod/.bashrc.d/99-java"
-# above, we are adding the sdkman init to .bashrc (executing sdkman-init.sh does that), because one is executed on interactive shells, the other for non-interactive shells (e.g. plugin-host)
-ENV GRADLE_USER_HOME=/workspace/.gradle/
+	&& source "$HOME/.sdkman/bin/sdkman-init.sh" \
+	&& sed -i -e 's/sdkman_selfupdate_enable=true/sdkman_selfupdate_enable=false/g' \
+	-e 's/sdkman_selfupdate_feature=true/sdkman_selfupdate_feature=false/g' "$HOME/.sdkman/etc/config" \
+	&& sdk install java ${JAVA_VERSION} \
+	&& sdk install java 17.0.3.fx-zulu \
```

Comment:
We support two Java chunk variants (11 and 17):
https://github.com/gitpod-io/workspace-images/blob/main/chunks/lang-java/chunk.yaml

We use the 11 chunk in workspace-full:
https://github.com/gitpod-io/workspace-images/blob/8b5d017c319bb8364d2ab327500b505d80686444/dazzle.yaml#L30, which lands here https://hub.docker.com/r/gitpod/workspace-full

We also use it in a dedicated image for 11:
https://github.com/gitpod-io/workspace-images/blob/8b5d017c319bb8364d2ab327500b505d80686444/dazzle.yaml#L139, which lands here https://hub.docker.com/r/gitpod/workspace-java-11

As you can see we use a separate 17 chunk to assemble the 17 combination:
https://github.com/gitpod-io/workspace-images/blob/8b5d017c319bb8364d2ab327500b505d80686444/dazzle.yaml#L144, which lands here https://hub.docker.com/r/gitpod/workspace-java-17

The change on line 17 is hard coding the Java variant, I get why (because you want both in the same image).

However, this will also add Java 17 to the Java 11 image: https://hub.docker.com/r/gitpod/workspace-java-11

In other words, this feels like a breaking change to me...I think we need more detailed testing, to ensure we're not breaking `workspace-full`, `workspace-java-11`, and `workspace-java-17`.

Reply:
In hindsight, it might make more sense to create a new, separate chunk, that we only use in workspace-full. We would call it "two chunk java", or something like that, and only use it in workspace-full.

Reply:
> However, this will also add Java 17 to the Java 11 image: https://hub.docker.com/r/gitpod/workspace-java-11

it is intended, without that Java in VS Code experience is bad now

Reply:
Thanks @akosyakov , that helps me understand what we're trying to accomplish. Let me look at this PR again, to see how testing was done, to see if I have any other questions. I looked last night, and it sounded like only one image was tested.